### PR TITLE
Update genymotion from 3.5.0 to 3.5.1

### PR DESCRIPTION
**Pull request by yurikoles**
Signed-off-by: Yurii Kolesnykov <root@yurikoles.com>

**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

### CODE REVIEW

**Comment by yurikoles** [comment]
My regex-fu is weak, so `livecheck` may be suboptimal.

File: Casks/g/genymotion.rb [hunk 1]
```diff
@@ -9,9 +9,11 @@
 
   livecheck do
     url "https://www.genymotion.com/download/"
-    regex(/Genymotion\s*Desktop\s*(\d+(?:\.\d+)+)/i)
+    regex(/genymotion-(\d+(?:\.\d+)+)\.dmg/i)
```
Comment (samford) [inline Casks/g/genymotion.rb:12]:
```suggestion
    regex(/href=.*?Genymotion[._-]v?(\d+(?:\.\d+)+)\.(?:dmg|pkg)/i)
```

This suggestion is a variation on our typical pattern for matching the version from a filename in an `href` attribute. When it comes to casks, I sometimes include alternative filename extensions if it's using something like `dmg`, as the regex may fail to match a new version in the future if upstream switches to a different file format (e.g., pkg, zip, etc.). Upstream is using only the file extension to distinguish between files for macOS and Windows, so I've opted to just use the macOS-specific file extensions (dmg, pkg) for now.

**Comment by yurikoles** [comment]
According to `otool` even `3.5.1` is built with deployment target `10.13`, thanks to [stackoverflow](https://stackoverflow.com/q/17143373/978568).
```sh
otool -l /Volumes/Genymotion\ Desktop/Genymotion.app/Contents/MacOS/genymotion | grep -E -A4 '(LC_VERSION_MIN_MACOSX|LC_BUILD_VERSION)'
      cmd LC_VERSION_MIN_MACOSX
  cmdsize 16
  version 10.13
      sdk 10.14
Load command 10
```

In [release notes](https://docs.genymotion.com/desktop/Get_started/012_Macos_install/) they state it was **_tested_** on Ventura and Sonoma. So I'm unsure does it actually work on pre-Ventura.

File: Casks/g/genymotion.rb (hunk 1 above)
Comment (yurikoles) [reply Casks/g/genymotion.rb]: Thank you for the explanation, I'll try to remember it.

Comment (yurikoles) [reply Casks/g/genymotion.rb]: Thank you for the explanation, I'll try to remember it.

**Review by bevanjkay — APPROVED** [review]

**Pull request merged**